nim_status/lib/migration.nim

```diff
@@ -0,0 +1,125 @@
+import sqlcipher, results
+import sequtils, tables, algorithm
+import stew/byteutils
+import migrations/sql_scripts
+import nimcrypto
+import chronicles
+
+type MigrationResult = Result[string, string]
+
+proc createMigrationTableIfNotExists*(db: DbConn) =
+  const settingSQL = """CREATE TABLE IF NOT EXISTS migrations (
+                          name VARCHAR NOT NULL PRIMARY KEY,
+                          hash VARCHAR NOT NULL
+                        )"""
+  db.execScript(settingSQL)
+
+
+proc getLastMigrationExecuted*(db: DbConn): MigrationResult =
+  db.createMigrationTableIfNotExists()
+  const query = "SELECT name FROM migrations ORDER BY rowid DESC LIMIT 1;"
+  for r in rows(db, query):
```

Comment:
This for seems useless since you `LIMIT 1`

Reply:
The api has changed, it should now be similar to:
```nim
let settings = db.one(Settings, query)
if not settings.isSome:
    raise newException(ValueError, "No records found for settings")	    raise newException(ValueError, "No record found for settings")
```